functions/src/resource.ts

```diff
@@ -286,16 +330,48 @@ export class S3ResourceObject extends BaseResourceObject {
     this.region = doc.region;
   }
 
-  apiResult(): S3ResourceObject {
-    const result = super.apiResult() as S3ResourceObject;
+  get publicPath() {
```

Comment:
Not necessary in this PR but it seems it would be helpful to put the S3ResourceObject class in its own file. I think it would make looking at the code easier.

Reply:
Yeap, I'll do it later not to add more diff to this PR.